### PR TITLE
Add autosave plugin

### PR DESCRIPTION
**Pull request by vincens2005**

### CODE REVIEW

**Review by franko — COMMENTED** [review]
Nice work, thank you for sharing. I just made a comment about the spawning of a new thread, you may have a closer look.

File: plugins/autosave.lua [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    end
+  end
+end
+
+
+local function updatepress()
+  -- set last keypress time to now
+  last_keypress = os.time()
+  -- put loop in coroutine so it doesn't lag out this script
+  core.add_thread(loop_for_save)
```
[the remark this answers is not in the document]
Comment (vincens2005) [reply plugins/autosave.lua:35]: good idea. fixing rn.

**Comment by francesco-st** [comment]
Thank you, I made a small fix to use self.filename in on_text_change instead of core.active_view:get_name().

I merged your plugin in the branch "lite-xl-1.16" so that they are visible for the next lite-xl release I am preparing. The method on_text_change is not available with the current release of lite-xl.

**Comment by francesco-st** [comment]
If you want to check your plugin is now in:

https://github.com/franko/lite-plugins/tree/lite-xl-1.16

**Comment by franko** [comment]
Closing the PR. The plugin is already in the [lite-xl-1.16 branch](https://github.com/franko/lite-plugins/tree/lite-xl-1.16). Thank you for the contribution.

**Comment by vincens2005** [comment]
> Thank you, I made a small fix to use self.filename in on_text_change instead of core.active_view:get_name().
> 
> I merged your plugin in the branch "lite-xl-1.16" so that they are visible for the next lite-xl release I am preparing. The method on_text_change is not available with the current release of lite-xl.

Cool. I was trying to find a way to get the filename but couldn't thanks for fixing that.

**Pull request closed without merging**